x-pack/legacy/plugins/apm/server/lib/transactions/get_transaction_by_trace/index.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  PROCESSOR_EVENT,
+  TRACE_ID,
+  PARENT_ID,
+  SERVICE_NAME,
+  TRACE,
+  TRANSACTION,
+  TIMESTAMP
+} from '../../../../common/elasticsearch_fieldnames';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { Setup } from '../../helpers/setup_request';
+import { ProcessorEvent } from '../../../../common/processor_event';
+
+export async function getTransactionByTraceId(traceId: string, setup: Setup) {
+  const { client, indices } = setup;
+  const params = {
+    index: indices['apm_oss.transactionIndices'],
+    body: {
+      _source: [TIMESTAMP, TRANSACTION, SERVICE_NAME, TRACE],
+      size: 1,
+      query: {
+        bool: {
+          should: [
+            {
+              constant_score: {
+                filter: {
+                  bool: {
+                    must_not: { exists: { field: PARENT_ID } }
+                  }
+                }
+              }
+            }
+          ],
+          filter: [
+            { term: { [TRACE_ID]: traceId } },
+            { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } }
+          ]
+        }
+      }
+    }
+  };
+
+  const resp = await client.search<Transaction>(params);
+  return {
+    transaction: resp.hits.hits[0]?._source
```

Comment:
If we end up cherry-picking individual field it's better to return them explicitly (currently the types says that a full `transaction` is returned but already now that's not correct afaict).